homeassistant/components/wyoming/satellite.py

```diff
@@ -0,0 +1,375 @@
+
+    @property
+    def is_running(self) -> bool:
+        """Return True if satellite is running."""
+        return self._is_running
+
+    @is_running.setter
+    def is_running(self, value: bool) -> None:
+        """Set whether satellite is running or not."""
+        self._is_running = value
+
+        # Unblock waiting for enabled
+        self._device_updated_event.set()
+
+    async def run(self) -> None:
+        """Run and maintain a connection to satellite."""
+        _LOGGER.debug("Running satellite task")
+        self._pipeline_id = pipeline_select.get_chosen_pipeline(
+            self.hass,
+            DOMAIN,
+            self.device.satellite_id,
+        )
+        self.is_enabled = self.device.is_enabled
+        remove_listener = self.device.async_listen_update(self._device_updated)
+
+        try:
+            while self.is_running:
+                try:
+                    if not self.is_enabled:
+                        await self._device_updated_event.wait()
+                        if not self.is_running:
+                            # Satellite was stopped while waiting to be enabled
+                            break
+
+                    await self._run_once()
+                except asyncio.CancelledError:
+                    raise
+                except Exception:  # pylint: disable=broad-exception-caught
+                    _LOGGER.exception(
+                        "Unexpected error running satellite. Restarting in %s second(s)",
+                        _RECONNECT_SECONDS,
+                    )
+                    await asyncio.sleep(_RESTART_SECONDS)
+        finally:
+            # Ensure sensor is off
+            if self.device.is_active:
+                self.device.set_is_active(False)
+
+            remove_listener()
+
+        _LOGGER.debug("Satellite task stopped")
+
+    def _device_updated(self, device: SatelliteDevice) -> None:
+        """Reacts to updated device settings."""
+        pipeline_id = pipeline_select.get_chosen_pipeline(
+            self.hass,
+            DOMAIN,
+            self.device.satellite_id,
+        )
+
+        stop_pipeline = False
+        if self._pipeline_id != pipeline_id:
+            # Pipeline has changed
+            self._pipeline_id = pipeline_id
+            stop_pipeline = True
+
+        if self.is_enabled and (not self.device.is_enabled):
+            stop_pipeline = True
+
+        self.is_enabled = self.device.is_enabled
+        self._device_updated_event.set()
+        self._device_updated_event.clear()
+
+        if stop_pipeline:
+            self._audio_queue.put_nowait(None)
+
+    async def _run_once(self) -> None:
+        """Run pipelines until an error occurs."""
+        if self.device.is_active:
+            self.device.set_is_active(False)
+
+        while True:
+            try:
+                await self._connect()
+                break
+            except ConnectionError:
+                _LOGGER.exception(
```

Comment:
Why log the exception stack trace here? It's an expected exception and no need to log stack.